sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3768,6 +3768,24 @@ class SQLConf extends Serializable with Logging {
     }
   }
 
+  /**
+   * Returns the [[Comparator]] for the current configuration,
+   * which can be used to compare two identifiers.
+   */
+  private[sql] def comparator: Comparator = {
+    if (caseSensitiveAnalysis) {
+      org.apache.spark.sql.catalyst.analysis.caseSensitiveComparator
+    } else {
+      org.apache.spark.sql.catalyst.analysis.caseInsensitiveComparator
+    }
+  }
+
+  private[sql] val fieldNameOrdering = new Ordering[String] {
```

Comment:
not sure how much difference it makes, however checking config (caseSensitiveAnalysis) for every evaluation doesn't look necessary, can we do create an ordering per config ...
```
  private[sql] def fieldNameOrdering: Ordering[String] = {
    if (caseSensitiveAnalysis) {
      (a: String, b: String) => a.compareTo(b)
    } else {
      (a: String, b: String) => a.compareToIgnoreCase(b)
    }
  }
```

Reply:
Thank you. It's reasonable.